pilot/pkg/xds/workload.go

```diff
@@ -51,22 +51,23 @@ func (e WorkloadGenerator) GenerateDeltas(
 		// Nothing changed..
 		return nil, nil, model.XdsLogDetails{}, false, nil
 	}
+
 	subs := sets.New(w.ResourceNames...)
-	addresses := updatedAddresses
-	// If it is not a wildcard, filter out resources we are not subscribed to
-	if !w.Wildcard {
-		addresses = addresses.Intersection(subs)
-	}
-	// Specific requested resource: always include
-	addresses = addresses.Merge(req.Delta.Subscribed)
-	addresses = addresses.Difference(req.Delta.Unsubscribed)
-	if !w.Wildcard {
-		// We only need this for on-demand. This allows us to subscribe the client to resources they
-		// didn't explicitly request.
-		// For wildcard, they subscribe to everything already.
-		// TODO: optimize me
-		additional := e.Server.Env.ServiceDiscovery.AdditionalPodSubscriptions(proxy, addresses, subs)
-		addresses.Merge(additional)
+	var addresses sets.String
+	if isReq {
+		// this is from request, we only send response for the subscribed address
```

Comment:
To make sure I understand: if the XDS request subscribes to "A", we send response only for "A". If a following XDS request subscribes to "B" - is the intent to send only the response to B (not A) ? 

And I assume if A has changed - an independent response will be sent ? 

May be worth adding few more lines of comment.

Reply:
>To make sure I understand: if the XDS request subscribes to "A", we send response only for "A". If a following XDS request subscribes to "B" - is the intent to send only the response to B (not A) ?

Yes

If A changed, we will trigger push differently behaves like endpoint update